cmd/pitr/recoverer/recoverer.go

```diff
@@ -0,0 +1,252 @@
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", errors.Wrap(err, "get output")
+	}
+	nodes := strings.Split(string(out), "node:")
+	sort.Strings(nodes)
+	lastHost := ""
+	for _, node := range nodes {
+		if strings.Contains(node, "wsrep_ready:ON:wsrep_connected:ON:wsrep_local_state_comment:Synced:wsrep_cluster_status:Primary") {
+			nodeArr := strings.Split(node, ".")
+			lastHost = nodeArr[0]
+		}
+	}
+	if len(lastHost) == 0 {
+		return "", errors.New("cant find host")
+	}
+
+	return lastHost + "." + r.pxcServiceName, nil
+}
+
+func (r *Recoverer) Run() error {
+	startGTID, err := r.GetLastBackupGTID()
+	if err != nil {
+		return errors.Wrap(err, "get last gtid")
+	}
+	err = r.GetBinlogList(startGTID)
+	if err != nil {
+		return errors.Wrap(err, "get binlog list")
+	}
+	err = r.Recover()
+	if err != nil {
+		return errors.Wrap(err, "recover")
+	}
+	return nil
+}
+
+func (r *Recoverer) DownloadBinlogs() error {
+	for _, binlog := range r.storage.ListObjects("") {
+		if strings.Contains(binlog, "set") {
+			continue
+		}
+		data, err := r.storage.GetObject(binlog)
+		if err != nil {
+			return errors.Wrap(err, "get object")
+		}
+		err = ioutil.WriteFile(binlog, data, 0644)
+		if err != nil {
+			return errors.Wrap(err, "write file")
+		}
+		r.binlogs = append(r.binlogs, binlog)
+	}
+	return nil
+}
+
+func (r *Recoverer) Recover() error {
+	host, err := r.getHost()
+	if err != nil {
+		return errors.Wrap(err, "get host")
+	}
+
+	for _, binlog := range r.binlogs {
+		err = r.DownloadBinlog(binlog)
+		if err != nil {
+			return errors.Wrap(err, "download binlog")
+		}
+		flags := ""
+
+		// TODO: add logic for all types
+		switch r.recoverType {
+		case Skip:
+			flags = "--exclude-gtids=" + r.excludingGTIDSet
+		case Transaction:
+		case Date:
+		case Latest:
+		}
+		cmdString := "mysqlbinlog /tmp/" + binlog + " " + flags + " | mysql -h" + host + " -u" + r.pxcUser + " -p$PXC_PASS"
+		cmd := exec.Command("sh", "-c", cmdString)
+		var outb, errb bytes.Buffer
+		cmd.Stdout = &outb
+		cmd.Stderr = &errb
+		err := cmd.Run()
+		if err != nil {
+			return errors.Wrap(err, "run cmd")
+		}
+		if errb.Bytes() != nil {
+			errors.Errorf("cmd error: %s", errb.String())
+		}
+		err = os.Remove("/tmp/" + binlog)
+		if err != nil {
+			return errors.Wrap(err, "remove binlog")
+		}
+	}
+
+	return nil
+}
+
+func (r *Recoverer) DownloadBinlog(binlog string) error {
+	content, err := r.storage.GetObject(binlog)
+	if err != nil {
+		return errors.Wrap(err, "get object")
+	}
+	f, err := os.Create("/tmp/" + binlog)
+	if err != nil {
+		return errors.Wrap(err, "create file")
+	}
+	defer f.Close()
+	_, err = f.Write(content)
+	if err != nil {
+		return errors.Wrap(err, "write content to file")
+	}
+	return nil
+}
+
+func (r *Recoverer) GetLastBackupGTID() (int64, error) {
+	sep := []byte("GTID of the last")
+
+	o, err := r.storage.GetObject(r.backupNAme + "/xtrabackup_info.lz4.00000000000000000000")
```

Comment:
is there any documentation on `/xtrabackup_info.lz4.00000000000000000000` format?
and why so strange name?